File: packages/dashcam_player/android/src/main/kotlin/com/dashcam/player/DashcamPlatformViewFactory.kt
package com.dashcam.player

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformViewFactory

/**
 * Factory that creates DashcamPlatformView instances when Flutter requests an AndroidView.
 */
class DashcamPlatformViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): DashcamPlatformView {
        return DashcamPlatformView(context, viewId)
    }
}

File: packages/dashcam_player/ios/Classes/DashcamPlatformViewFactory.swift
import Flutter

/// Factory that creates DashcamPlatformView instances for Flutter.
class DashcamPlatformViewFactory: NSObject, FlutterPlatformViewFactory {

    func create(withFrame frame: CGRect,
                viewIdentifier viewId: Int64,
                arguments args: Any?) -> FlutterPlatformView {
        return DashcamPlatformView(frame: frame, viewId: viewId)
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}
